App.tsx

```diff
@@ -1,59 +1,50 @@
-import {useColorScheme} from 'react-native';
+import React, {useState, useEffect} from 'react';
+import {View, Text} from 'react-native';
+import auth, {type FirebaseAuthTypes} from '@react-native-firebase/auth';
+import {NavigationContainer} from '@react-navigation/native';
 
-import {FavouritesProvider} from './providers/FavouritesProvider';
-import HomeScreen from './screens/HomeScreen';
-import DetailedMuseumScreen from './screens/DetailedMuseumScreen';
-import type Museum from './model/Museum';
-import {getAppColors} from './utils/colors';
-import {getAppStyles} from './utils/styles';
+import UnAuthNavigator from './navigators/UnAuthNavigator';
+import AuthNavigator from './navigators/AuthNavigator';
+import {AuthProviders} from './providers/AuthProviders';
 
-export type MainRootStackParamList = {
-  Home: undefined;
-  DetailedMuseum: {museum: Museum};
-};
+type NullableUser = FirebaseAuthTypes.User | null;
 
-export type MainStackNavigation = NavigationProp<MainRootStackParamList>;
+export default function App() {
+  // Set an initializing state whilst Firebase connects
+  const [initializing, setInitializing] = useState(true);
+  const [user, setUser] = useState<NullableUser>();
 
-const Stack = createNativeStackNavigator<MainRootStackParamList>();
+  // Handle user state changes
+  function onAuthStateChanged(user: NullableUser) {
+    setUser(user);
+    if (initializing) setInitializing(false);
+  }
 
-export default function App() {
-  return (
-    <FavouritesProvider>
-      <MainNavigator />
-    </FavouritesProvider>
-  );
-}
+  useEffect(() => {
+    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    return subscriber; // unsubscribe on unmount
+  }, []);
+
+  if (initializing)
+    return (
+      <View>
+        <Text>Loading ...</Text>
+      </View>
+    );
 
-function MainNavigator() {
-  const colorScheme = useColorScheme();
-  const {backgroundStyle} = getAppStyles(colorScheme);
-  const {backgroundColor, headerColor} = getAppColors(colorScheme);
+  if (!user) {
+    return (
+      <NavigationContainer>
+        <UnAuthNavigator />
+      </NavigationContainer>
+    );
+  }
 
   return (
-    <NavigationContainer>
-      <Stack.Navigator
-        screenOptions={() => ({
-          tabBarStyle: backgroundStyle,
-          headerStyle: {backgroundColor: backgroundColor},
-          headerTintColor: headerColor,
-          headerBackTitleVisible: false,
-        })}>
-        <Stack.Screen
-          name="Home"
-          component={HomeScreen}
-          options={{headerShown: false}}
-        />
-        <Stack.Screen
-          name="DetailedMuseum"
-          component={DetailedMuseumScreen}
-          options={({route}) => ({title: route.params.museum.title})}
-        />
-      </Stack.Navigator>
-    </NavigationContainer>
+    <AuthProviders>
+      <NavigationContainer>
+        <AuthNavigator />
+      </NavigationContainer>
+    </AuthProviders>
```

Comment:
I think I would rather try to only switch navigators while logging in, this way we don't have to reinstantiate the whole navigation, and the code is simpler :) 

```suggestion
    <AuthProviders>
      <NavigationContainer>
        {user ? <AuthNavigator /> : <UnAuthNavigator />
      </NavigationContainer>
    </AuthProvid

Reply:
The reaso why i did this is because of auth providers, right now there is no difference is it above UnAuthNavigator but later it might be. If you have sugesstion how to solve it I appreciate it.

Reply:
In this case only the `AuthNavigator` could be wrapped by `AuthProviders` - it could already done in the file where the `AuthNavigator` is declared

Reply:
But Auth provider is not a Navigator, NavigatorContainer will complain about it

Reply:
You are a wizard!
Previusly i was trying to do that, i got error, right now everyting works perfectly!